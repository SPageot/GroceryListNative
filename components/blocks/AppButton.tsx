import { Button } from "react-native";
import React from "react";
import styled from "styled-components";
import { SubmitType } from "../../types/types";

const AppActionButton = styled(Button)`
  height: 100%;
  width: 20px;
  color: #fff;
  background-color: navy;
`;

const AppButton = ({ title, onPress, color }: SubmitType) => {
  return <AppActionButton color={color} title={title} onPress={onPress} />;
};

export { AppButton };
