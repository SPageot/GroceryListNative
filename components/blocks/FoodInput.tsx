import { View, Text, TextInput } from "react-native";
import React from "react";
import styled from "styled-components";
import { FoodInputType } from "../../types/types";

const FoodInputName = styled(TextInput)`
  height: 40px;
  width: 75%;
  border: 1px solid black;
`;

const FoodInput = ({
  placeholder,
  onChangeText,
  defaultValue,
}: FoodInputType) => {
  return (
    <FoodInputName
      placeholder={placeholder}
      onChangeText={onChangeText}
      defaultValue={defaultValue}
    />
  );
};

export { FoodInput };
