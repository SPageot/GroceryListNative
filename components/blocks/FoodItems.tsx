import _ from "lodash";
import React from "react";
import { ScrollView, Text, View } from "react-native";
import { Swipeable } from "react-native-gesture-handler";
import styled from "styled-components";
import { FoodItemType } from "../../types/types";
import { AppButton } from "./AppButton";

const FoodsContainer = styled(ScrollView)`
  height: 90%;
  width: 100%;
  background-color: #0077b6;
`;
const DeleteContainer = styled(View)`
  justify-content: center;
  background-color: red;
`;

const FoodItemContainer = styled(View)`
  height: 80px;
  width: 100%;
  padding: 10px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: blue;
`;

const FoodItem = styled(Text)`
  color: #fff;
  font-size: 17px;
  width: 80%;
`;

const FoodItems = ({ foodItemsArray, onPress }: FoodItemType) => {
  const swipeRightAction = (item: string) => {
    return (
      <DeleteContainer>
        <AppButton color="#fff" title="Delete" onPress={() => onPress(item)} />
      </DeleteContainer>
    );
  };

  return (
    <FoodsContainer>
      {!_.isEmpty(foodItemsArray)
        ? foodItemsArray.map((item, i) => {
            return (
              <Swipeable
                renderRightActions={() => swipeRightAction(item)}
                key={i}
              >
                <FoodItemContainer>
                  <FoodItem>{item}</FoodItem>
                </FoodItemContainer>
              </Swipeable>
            );
          })
        : null}
    </FoodsContainer>
  );
};

export default FoodItems;
