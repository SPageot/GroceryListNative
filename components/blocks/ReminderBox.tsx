import { View, Text, Pressable, Keyboard } from "react-native";
import React from "react";
import styled from "styled-components";
import { Input } from "../form/Input";
import { AppButton } from "./AppButton";
import { ReminderProps } from "../../types/types";

const ReminderBoxContainer = styled(View)`
  height: 100%;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  justify-content: center;
  align-items: center;
  z-index: 1000;
`;

const ReminderInfoBox = styled(Pressable)`
  height: 470px;
  width: 95%;
  background-color: #191970;
  justify-content: space-evenly;
  border-radius: 35px;
`;

const ReminderBoxHeaderContainer = styled(View)`
  width: 100%;
  align-items: center;
`;

const ReminderBoxHeader = styled(Text)`
  color: #fff;
  font-size: 20px;
  font-weight: bold;
`;

const ReminderHeaderContainer = styled(View)`
  width: 100%;
  align-items: center;
`;

const ReminderMessageContainer = styled(View)`
  width: 100%;
  align-items: center;
`;

const ReminderOptionsContainer = styled(View)`
  width: 100%;
  flex-direction: row;
  justify-content: space-around;
`;

const ReminderBox = ({
  handleHeaderChangeText,
  handleChangeText,
  handleCancelPress,
  handleSubmitPress,
  headerValue,
  messageValue,
}: ReminderProps) => {
  return (
    <ReminderBoxContainer>
      <ReminderInfoBox onPress={Keyboard.dismiss}>
        <ReminderBoxHeaderContainer>
          <ReminderBoxHeader>Reminder</ReminderBoxHeader>
        </ReminderBoxHeaderContainer>
        <ReminderHeaderContainer>
          <Input
            type="default"
            onChangeText={handleHeaderChangeText}
            defaultValue={headerValue}
            placeholder="Enter header text..."
          />
        </ReminderHeaderContainer>
        <ReminderMessageContainer>
          <Input
            type="reminders"
            onChangeText={handleChangeText}
            defaultValue={messageValue}
            placeholder="Enter reminder message..."
          />
        </ReminderMessageContainer>
        <ReminderOptionsContainer>
          <AppButton title="Cancel" onPress={handleCancelPress} color="#fff" />
          <AppButton title="Submit" onPress={handleSubmitPress} color="#fff" />
        </ReminderOptionsContainer>
      </ReminderInfoBox>
    </ReminderBoxContainer>
  );
};

export { ReminderBox };
