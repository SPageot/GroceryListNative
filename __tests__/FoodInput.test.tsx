import React from "react";
import { FoodInput } from "../components/blocks/FoodInput";
import TestRenderer from "react-test-renderer";

describe("Food Input", () => {
  const mockFunction = jest.fn();
  const mockFoodInput = TestRenderer.create(
    <FoodInput
      onChangeText={mockFunction}
      defaultValue="value"
      placeholder="value"
    />
  );
  it("should render correctly", () => {
    expect(mockFoodInput.toJSON()).toMatchSnapshot();
  });
});
