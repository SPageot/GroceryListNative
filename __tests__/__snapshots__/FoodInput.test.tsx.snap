// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Input should render correctly 1`] = `
<TextInput
  className="sc-beqWaB dZHoAW"
  defaultValue="value"
  onChangeText={[MockFunction]}
  placeholder="value"
/>
`;
